anybox/recipe/odoo/vcs/git.py

```diff
@@ -406,6 +413,26 @@ def update_fetched_branch(self, branch):
                     self.log_call(['git', 'reset', '--hard', 'FETCH_HEAD'],
                                   callwith=update_check_call)
 
+    def _no_edit(self, cmd):
+        if self.git_version >= (1, 7, 10):
+            # --edit and --no-edit appear with Git 1.7.10
+            # see Documentation/RelNotes/1.7.10.txt of Git
+            # (https://git.kernel.org/cgit/git/git.git/tree)
+            cmd.insert(2, '--no-edit')
+        return cmd
+
+    def offline_merge(self, revision):
+        """Merge revision into current branch"""
+        if ishex(revision):
+            if not self.has_commit(revision):
+                raise UserError("Commit %s not found in git repository "
+                                "%s (offline mode)" % (revision, self))
+        elif not self._is_a_branch(revision):
+            raise UserError("Branch %s not found in git repository "
+                            "%s (offline mode)" % (revision, self))
+        cmd = self._no_edit(['git', 'merge', revision])
+        self.log_call(cmd)
```

Comment:
This method does not manage its working directory like the `merge` method does, so offline merging still fails with something like this

```
anybox.recipe.odoo.vcs.git: /home/odoo/odoo/parts/server-ux> call ['git', 'merge', '--no-edit', '5b8d5be']
merge: 5b8d5be - not something we can merge
```
Current working dir is `parts` at that point, so it makes sense.

This fixes the problem for me:
```
        with working_directory_keeper:
            os.chdir(self.target_dir)
            self.log_call(cmd)
```